Rakefile

```diff
@@ -1,12 +1,28 @@
 require 'rake'
 require 'rspec/core/rake_task'
 require 'puppetlabs_spec_helper/rake_tasks'
+require 'puppet'
+
 begin
   require 'beaker/tasks/test' unless RUBY_PLATFORM =~ /win32/
 rescue LoadError
   #Do nothing, only installed with system_tests group 
 end
 
+
+# If puppet does not support symlinks (e.g., puppet <= 3.5) we cannot use
+# puppetlabs_spec_helper's `rake spec` task because it requires symlink
+# support. Redine `rake spec` without the symlink call and restrict the pattern
+# match to files under the 'unit' directory.
+if !!File::ALT_SEPARATOR and !Puppet::FileSystem.respond_to?(:symlink)
```

Comment:
@glennsarti I chatted with @Iristyle and it sounds like we want to avoid `Puppet.features`, but could instead rely on `Puppet::Util::Platform.windows?`. That actually also just checks for [ALT_SEPARATOR](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/platform.rb#L10) under the hood, but at least someone coming along later will have some idea of what we're doing. I've updated the PR along those lines.